### PR TITLE
[feat] - Add macos/setup-cyclaw-keys.sh for Apple Silicon key bootstrap

### PR DESCRIPTION
**Pull request by cgfixit**
## Proposed changes

`setup-guide.md` still tells operators to `export CYCLAW_API_KEY="$(openssl rand -hex 20)"` and (worse) to paste secrets into `~/.zshrc`. The installer deliberately does **not** set keys, and `invoke-cyclaw.sh` only warns when `CYCLAW_API_KEY` is missing. There was no first-class macOS script that generates the local key the docs already specify, prompts for operator-supplied tokens CyClaw cannot invent, persists them in the stores the code actually reads (Keychain + gitignored dotenv), and never inlines a secret into a profile file, plist, or the `cyclaw` shim.

`macos/setup-cyclaw-keys.sh` (Darwin / arm64 only; bash 3.2; BSD userland; `[cyclaw]` prefix — same contract as the rest of `macos/`).

| Env var | How | Keychain service |
|---|---|---|
| `CYCLAW_API_KEY` | Autogenerated (`openssl rand -hex 20`) | `com.cgfixit.cyclaw.api-key` |
| `TELEGRAM_BOT_TOKEN` | Prompt, skip allowed | `com.cgfixit.cyclaw.telegram-bot-token` |
| `ANTHROPIC_API_KEY` | Prompt, skip allowed | `com.cgfixit.cyclaw.anthropic-api-key` |
| `GROK_API_KEY` | Prompt, skip allowed, or `--grok-dummy` | `com.cgfixit.cyclaw.grok-api-key` |
| `GH_TOKEN` (+ `GITHUB_TOKEN`) | Prompt, skip allowed | `com.cgfixit.cyclaw.gh-token` |

Persistence, in order:

1. **Keychain** — official path. `security` is invoked with a bare `-w` so the secret is never on argv (same contract as `cyclaw-keychain-set.sh`; generated values go through a 0600 tempfile + `/usr/bin/expect`). launchd still injects via `cyclaw-keychain-env.sh`.
2. **`$CYCLAW_HOME/.env`** (default `~/.CyClaw/.env`) — `chmod 600`, gitignored.
3. **Checkout `.env`** — only if a repo is found (`--repo-path` / `$CYCLAW_REPO` / `~/.CyClaw/repo`).
4. **rc source block** — `# >>> cyclaw keys >>>` sources the same dotenv this script wrote. Secrets are never written into `~/.zshrc` / `~/.bash_profile`.

Claude is `ANTHROPIC_API_KEY` because that is the only name `llm/client.py` reads (`CLAUDE_API_KEY` is a docs leftover).

Flags: `--rotate`, `--skip-prompts`, `--grok-dummy`, `--no-keychain`, `--no-env-file`, `--no-repo-env`, `--no-profile-edit`, `--no-print-key`, `--print-key`, `--repo-path`.

`uninstall-cyclaw.sh` now strips the keys rc block alongside the existing path/function blocks. Home `.env` is still kept unless `--remove-home`.

Follow-up on this branch (`2ef9204`) also closes the CI failure and the Codex P1/P2 items: fake Keychain stdin log is service-keyed; rc follows effective `CYCLAW_HOME`; INT/TERM restore echo and exit 130/143; `_env_get` decodes POSIX `'\''`.

**Invariant / Governance Impact**
- None of the six security invariants or I6 module isolation are affected. This is installer/glue under `macos/` — `gate.py`, `graph.py`, and `mcp_hybrid_server.py` never import it (I6).
- RAG-first entry, topology=policy, triple-gated fallback (I3), audit convergence, and soul governance (I5) are untouched.
- Evidence: no graph/gate/soul/audit/retrieval edits; `verify-skills (invariant-guard)` and `invariant-guard` both pass on this head.

---

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation Update (if none of the other choices apply)
- [ ] Invariant / Governance refinement (use this for changes that strengthen or evolve the 6 invariants, I6 isolation, or harness phases)

**Optional free-text scope note:** Out-of-band installer / macOS glue (`macos/`, `tests/test_setup_cyclaw_keys.py`, `macos/README.md`, `setup-guide.md`). Not a core graph/gate/soul path.

---

## Benefits / why

- Operators get a first-class Apple Silicon path that matches what the code actually reads (Keychain + gitignored dotenv) instead of pasting secrets into `~/.zshrc`.
- Secrets never land in a profile file, plist, or the `cyclaw` shim — improves offline/air-gapped operational hygiene without changing runtime policy.
- `--skip-prompts --grok-dummy` is the offline/pytest path; generated `CYCLAW_API_KEY` can be printed once for the Soul / operator console (`--no-print-key` exists).
- Uninstall now removes the keys rc block, so a later `uninstall-cyclaw.sh` does not leave a dangling source of a deleted dotenv.

---

## Risks to monitor

- **One-time stdout print of `CYCLAW_API_KEY`.** Default when a new key is generated (matches `setup-guide.md`). Scrollback / screenshot exposure is the residual. `--no-print-key` exists. Do not paste that line into an issue, chat log, or commit.
- **Checkout `.env` is gitignored (`*.env`) but a later `git add -f` would publish keys.** Same residual as any dotenv workflow. Tests default `--no-repo-env` so CI does not write the checkout `.env`.
- **Re-run with a changed `CYCLAW_HOME` does not rewrite an existing healthy keys marker block.** First install writes the correct source path; a later custom home leaves the old rc body in place. Detect by opening a new shell and checking whether the intended dotenv is sourced. Compensating: `--no-profile-edit` plus a hand edit, or delete the marker block and re-run.
- **Ctrl-C / TERM during a prompt exits 130 / 143 after the API key is already persisted.** Later tokens are not written. Expected; do not treat a leftover `CYCLAW_API_KEY` after abort as a leak of Telegram/Grok/GH tokens.
- Does **not** introduce a shortcut around audit convergence, weaken RAG-first, add network assumptions, touch subprocess isolation, or change soul evolution. No new audited free-text field.
- Drift detection: `tests/test_setup_cyclaw_keys.py` (16 tests, fake `security`, no real Keychain) plus `bash -n macos/setup-cyclaw-keys.sh`.

---

## Checklist

- [x] I have read the latest `docs/CyClaw Architecture Guide` (and any relevant Phase docs) and `SECURITY.md`
- [x] This change preserves all 6 security invariants and I6 module isolation (explicit evidence or invariant matrix included for core changes)
- [x] Full sandbox validation has been run (`cyclaw-sandbox-validator` or equivalent pytest + smoke tests on core RAG/agentic paths) and passes with no regressions
- [x] No new external network dependencies or mandatory online LLM assumptions were introduced without explicit justification + offline fallback path
- [x] For any agentic/fsconnect/harness change: two-phase audit, quota enforcement, governed delete/trash, and write guards have been verified
- [x] Relevant architecture docs, threat model notes, or harness phase documentation have been updated if core behavior or topology changed
- [x] Commit messages follow the title prefix convention above
- [x] For large or complex changes: before/after invariant matrix + sandbox evidence is included in "Further comments" or linked

---

## Further comments

Out-of-band installer glue. No change to graph topology, entry points, audit hashing, or soul governance.

**ELI5:** this script puts your CyClaw keys in the Mac Keychain and a locked `.env` file, and teaches new terminal tabs to load that file. It never pastes the actual secret into `.zshrc`.

**Privacy (cyclaw-advisor — advisory only, not licensed counsel):** no new personal-data store, no new audited free-text field (`utils/logger.py` still SHA-256-hashes `query`), I3 fallback forwarding still `send_local_context_to_grok/claude: false`, auth/memory subsystems untouched. No Critical/High. No escalation.

**Run**

```bash
bash macos/setup-cyclaw-keys.sh
source ~/.CyClaw/.env    # this tab; new tabs inherit via rc
```

### CODE REVIEW

**Comment by cgfixit** [comment]
> 
> ### 💡 Codex Review
> 
> Here are some automated review suggestions for this pull request.
> 
> **Reviewed commit:** `0a3db0df29`
>     
> 
> <details> <summary>ℹ️ About Codex in GitHub</summary>
> <br/>
> 
> [Your team has set up Codex to review pull requests in this repo](https://chatgpt.com/codex/cloud/settings/general). Reviews are triggered when you
> - Open a pull request for review
> - Mark a draft as ready
> - Comment "@codex review".
> 
> If Codex has suggestions, it will comment; otherwise it will react with 👍.
> 
> 
> 
> 
> Codex can also answer questions or update the PR. Try commenting "@codex address that feedback".
>             
> </details>

@codex apply fixes and fix ci that failed too

**Comment by cgfixit** [comment]
> 
> ### 💡 Codex Review
> 
> Here are some automated review suggestions for this pull request.
> 
> **Reviewed commit:** `0a3db0df29`
>     
> 
> <details> <summary>ℹ️ About Codex in GitHub</summary>
> <br/>
> 
> [Your team has set up Codex to review pull requests in this repo](https://chatgpt.com/codex/cloud/settings/general). Reviews are triggered when you
> - Open a pull request for review
> - Mark a draft as ready
> - Comment "@codex review".
> 
> If Codex has suggestions, it will comment; otherwise it will react with 👍.
> 
> 
> 
> 
> Codex can also answer questions or update the PR. Try commenting "@codex address that feedback".
>             
> </details>

@codex review and apply fixes and fix the currently failing ci(s) that failed too

**Comment by cgfixit** [comment]
## CI + review fix (`2ef9204`)

All three red jobs (`ubuntu-latest`, `ci (3.12)`, `verify-skills (CyClaw-Sandbox)`) were the **same test**, not three bugs:

`tests/test_setup_cyclaw_keys.py::test_skip_prompts_generates_key_into_home_env`

`--grok-dummy` stores two Keychain secrets. The fake `security` stub overwrote `FAKE_SECURITY_STDIN_LOG` on every `add-generic-password`, so the assertion read `dummy` instead of the generated `CYCLAW_API_KEY`.

Codex flagged this as P1 and claimed a follow-up commit (`56c4c90`) — that commit never landed on this branch.

### What landed

| Item | Severity | Fix |
|---|---|---|
| Fake Keychain stdin log clobber (the CI fail) | P1 | Append `service\\tsecret` per write; assert API key and dummy independently |
| rc always sourced `~/.CyClaw/.env` | P2 | Source block follows effective `CYCLAW_HOME` (portable `$HOME` form for the default) |
| Ctrl-C / TERM during a prompt continued writing | P2 | Restore echo and exit 130 / 143 |
| Apostrophe in an existing key 401'd on keep | P2 | `_env_get` decodes POSIX `'\\''` |
| Half-written keys marker | extra | Fail closed; leave the rc file untouched |
| Tests wrote a checkout `.env` | extra | `_run` defaults to `--no-repo-env` (`_SCRIPT_DIR/..` always looks like a repo) |

16 tests, including PTY SIGINT/SIGTERM. `ruff check --select E,F,I,B,C4,UP,S` clean on the test file. `bash -n` clean.

### Privacy (cyclaw-advisor) — advisory only, not licensed counsel

**Assessment:** this PR does not weaken CyClaw's existing data-handling controls. It is a local operator secret-bootstrap path; it does not create a new personal-data store, audit field, or off-box transfer.

**Citations / mechanics**
- No new audited free-text field (`utils/logger.py` query-hash invariant untouched).
- Secrets never written to `config.yaml`, plists, or the `cyclaw` shim — Keychain + `chmod 600` dotenv only.
- `step()` logs service / variable **names**, never values.
- rc file sources the dotenv; it does not inline secrets (GDPR Art. 32 / CCPA §1798.150-adjacent: reasonable security, not a new processing purpose).
- `GH_TOKEN` is an operator credential, stored locally. No email/IP collection. Auth store / memory subsystem unchanged.
- I3 fallback forwarding defaults still off.

**Flags**
- *Medium* — generated `CYCLAW_API_KEY` is printed once on purpose (`--no-print-key` exists). Scrollback / screenshot exposure is the residual. Do not paste that line into an issue, chat log, or commit.
- *Medium* — checkout `.env` is gitignored (`*.env`) but a later `git add -f` would publish keys. Same residual as any dotenv workflow.

**No Critical/High.** No escalation to outside counsel. No DSR / breach-notification angle — no end-user personal data is collected or moved.

CI on `2ef9204` should flip the three red jobs. Will watch the new run.

File: tests/test_setup_cyclaw_keys.py [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+    match = re.search(r"export CYCLAW_API_KEY='([0-9a-f]{40})'", text)
+    assert match, text
+    generated = match.group(1)
+    assert generated not in result.stderr
+    assert generated not in result.stdout
+    logged = argv_log.read_text(encoding="utf-8")
+    assert "-s com.cgfixit.cyclaw.api-key" in logged
+    assert "-T /usr/bin/security" in logged
+    assert generated not in logged
+    assert stdin_log.read_text(encoding="utf-8") == generated
```
[the remark this answers is not in the document]
Comment (cgfixit) [reply tests/test_setup_cyclaw_keys.py]: Fixed in `2ef9204`. Fake `security` now appends `service<TAB>secret` per write. `test_skip_prompts_generates_key_into_home_env` asserts the generated API key and `dummy` independently — that was the ubuntu / ci(3.12) / CyClaw-Sandbox failure.

File: macos/setup-cyclaw-keys.sh [hunk 2]
```diff
@@ -0,0 +1,557 @@
...
+  if grep -qF "$marker" "$rc_file" 2>/dev/null; then
+    step "rc already sources ~/.CyClaw/.env ($rc_file)"
+    return 0
+  fi
+  {
+    echo ""
+    echo "$marker"
+    echo "# Loads ~/.CyClaw/.env (chmod 600). Secrets are not stored in this rc file."
+    echo "if [ -f \"\$HOME/.CyClaw/.env\" ]; then"
+    echo "  . \"\$HOME/.CyClaw/.env\""
```
[the remark this answers is not in the document]
Comment (cgfixit) [reply macos/setup-cyclaw-keys.sh]: Fixed in `2ef9204`. Default home still sources `$HOME/.CyClaw/.env` (portable). A custom `CYCLAW_HOME` is resolved, `reject_shell_metachars`'d, and single-quoted into the rc block. Covered by `test_custom_cyclaw_home_rc_sources_that_env`.

File: macos/setup-cyclaw-keys.sh [hunk 3]
```diff
@@ -0,0 +1,557 @@
...
+  fi
+  if [ -n "$present" ]; then
+    printf '[cyclaw] %s (%s) already set via %s. Enter a new value to replace, or press return to keep: ' \
+      "$label" "$env_name" "$present" >&2
+  else
+    printf '[cyclaw] %s (%s) — paste the secret, or press return to skip: ' \
+      "$label" "$env_name" >&2
+  fi
+  # -s: no echo. bash 3.2 supports it. Restore echo on any exit path.
+  trap 'stty echo 2>/dev/null || true' EXIT INT TERM
```
[the remark this answers is not in the document]
Comment (cgfixit) [reply macos/setup-cyclaw-keys.sh]: Fixed in `2ef9204`. INT restores echo and exits 130; TERM restores echo and exits 143. PTY tests: `test_sigint_during_prompt_exits_130`, `test_sigterm_during_prompt_exits_143`. Already-written `CYCLAW_API_KEY` (generated before the first prompt) is kept; later tokens are not written.

File: macos/setup-cyclaw-keys.sh [hunk 4]
```diff
@@ -0,0 +1,557 @@
...
+# Only used to sync an existing .env into Keychain. Never echoed.
+_env_get() {
+  local file="$1" key="$2" line
+  [ -f "$file" ] || return 1
+  line="$(grep -E "^[[:space:]]*(export[[:space:]]+)?${key}=" "$file" | tail -n 1)" || return 1
+  line="${line#*=}"
+  # Strip one matching pair of single quotes if present.
+  case "$line" in
+    \'*\') line="${line#\'}"; line="${line%\'}" ;;
+    \"*\") line="${line#\"}"; line="${line%\"}" ;;
```
[the remark this answers is not in the document]
Comment (cgfixit) [reply macos/setup-cyclaw-keys.sh]: Fixed in `2ef9204`. `_env_get` now runs `_env_unquote`, which inverts `_shell_single_quote`'s POSIX `'\''` encoding. `test_keep_round_trips_apostrophe_in_existing_key` covers the keep-without-rotate path.

**Comment by cgfixit** [comment]
## CI re-check on `2ef9204`

The three previously red jobs are **green** on this head ([docs: required checks must pass on the head SHA](https://docs.github.com/en/pull-requests/reference/status-checks)):

| Check | Result |
|---|---|
| `ubuntu-latest` | success |
| `ci (3.12)` | success |
| `verify-skills (CyClaw-Sandbox)` | success (`verify.sh` + `smoke.sh`) |
| `macos-latest` | success |
| Codex P1/P2 threads | resolved |
| PR template (`Risks to monitor` + title prefix) | filled; advisory check now success |

`windows-latest` was still in the pytest/coverage step when this was posted — that matrix skips `tests/test_setup_cyclaw_keys.py` (`os.name == "nt"`). Not one of the original failures.

No further code change on this pass: pushing would restart already-green required checks for an edge-case (existing keys rc block not rewritten if `CYCLAW_HOME` later changes). That residual is documented in **Risks to monitor**.

File: macos/setup-cyclaw-keys.sh [hunk 5]
```diff
@@ -0,0 +1,616 @@
...
+
+  # Fail closed on a half-written marker block: never append a second copy
+  # and never silently treat a corrupt block as "already installed."
+  if grep -qxF "$KEYS_START" "$rc_file" 2>/dev/null; then
+    if ! grep -qxF "$KEYS_END" "$rc_file" 2>/dev/null; then
+      echo "[cyclaw] malformed cyclaw keys block in $rc_file (start without end); left unchanged" >&2
+      return 1
+    fi
+    step "rc already sources the keys dotenv ($rc_file)"
+    return 0
```
[the remark this answers is not in the document]
Comment (cgfixit) [reply macos/setup-cyclaw-keys.sh:438]: @codex review and apply fixes

File: macos/setup-cyclaw-keys.sh [hunk 6]
```diff
@@ -0,0 +1,616 @@
...
+    present=".env"
+  elif [ -n "$(eval "printf '%s' \"\${$env_name:-}\"")" ]; then
+    present="environment"
+  fi
+  if [ -n "$present" ]; then
+    printf '[cyclaw] %s (%s) already set via %s. Enter a new value to replace, or press return to keep: ' \
+      "$label" "$env_name" "$present" >&2
+  else
+    printf '[cyclaw] %s (%s) — paste the secret, or press return to skip: ' \
+      "$label" "$env_name" >&2
```
[the remark this answers is not in the document]
Comment (cgfixit) [reply macos/setup-cyclaw-keys.sh:476]: @codex review and apply fixes
